machine/v1beta1/types_azureprovider.go

```diff
@@ -174,6 +174,13 @@ type Image struct {
 	Version string `json:"version"`
 	// ResourceID specifies an image to use by ID
 	ResourceID string `json:"resourceID"`
+	// ThirdPartyImage indicates the image is published by a third party publisher and a Plan
+	// will be generated for it.
+	// +optional
+	ThirdPartyImage bool `json:"thirdPartyImage,omitempty"`
+	// SharedGallery specifies an image to use from an Azure Shared Image Gallery
+	// +optional
+	SharedGallery *AzureSharedGalleryImage `json:"sharedGallery,omitempty"`
```

Comment:
Is this related to the ThirdPartyImage?  Are they specified together or just coincidental in this PR?

Is there a distinction between nil and empty? If there isn't, the configuration API (distinct from workload API) is more discoverable and less prone to error by not being a pointer.

Reply:
> Is this related to the ThirdPartyImage? Are they specified together or just coincidental in this PR?

`ThirdPartyImage` is only related to Marketplace Images. There are [three types of images in the upstream API.](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/d2b4e8ed2313adc270a0911d9d0c64e344f34cc4/api/v1beta1/types.go#L316-L398) Our API has two images types (ID & marketplace) flattened together and this is adding the third as a separate struct. I'm open to other approaches, but I believe we would have to deprecate fields.

Based on our current use case there is no distinction between nil & empty, so I'm fine with removing the pointer.